test/smoke_test/bcda_client.go

```diff
@@ -125,13 +129,34 @@ func get(location string) *http.Response {
 
 func writeFile(resp *http.Response, filename string) {
 	defer resp.Body.Close()
+
+	var isGZIP bool
+	for _, h := range resp.Header.Values("Content-Encoding") {
+		if h == "gzip" {
+			isGZIP = true
+			break
+		}
+	}
+
+	if !isGZIP {
+		panic("Data responses should be gzip-encoded.")
+	}
+
+	reader, err := gzip.NewReader(resp.Body)
+	if err != nil {
+		panic(err)
+	}
+	defer reader.Close()
+
 	/* #nosec */
 	out, err := os.Create(filename)
 	if err != nil {
 		panic(err)
 	}
 	defer utils.CloseFileAndLogError(out)
-	num, err := io.Copy(out, resp.Body)
+
+	/* #nosec - not concerned with OOM since we're using this in a controlled setting */
```

Comment:
Do we want to have a general `nosec` here, or maybe something more specific so we're only skipping a single rule?  Example can be found [here](https://github.com/CMSgov/bcda-app/blob/master/bcda/models/postgres/repository.go#L98) with documentation [here](https://github.com/securego/gosec#annotating-code).